src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -0,0 +1,491 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // Contains test methods that can be used for FileInfo, DirectoryInfo, File or Directory.
+    public abstract class BaseSymbolicLinks_FileSystem : BaseSymbolicLinks
+    {
+        /// <summary>Creates a new file or directory depending on the implementing class.
+        /// If createOpposite is true, creates a directory if the implementing class is for File or FileInfo, or
+        /// creates a file if the implementing class is for Directory or DirectoryInfo.</summary>
+        protected abstract void CreateFileOrDirectory(string path, bool createOpposite = false);
+
+        protected abstract void AssertIsCorrectTypeAndDirectoryAttribute(FileSystemInfo linkInfo);
+
+        protected abstract void AssertLinkExists(FileSystemInfo linkInfo);
+
+        /// <summary>Calls the actual public API for creating a symbolic link.</summary>
+        protected abstract FileSystemInfo CreateSymbolicLink(string path, string pathToTarget);
+
+        private void CreateSymbolicLink_Opposite(string path, string pathToTarget)
+        {
+            Type t = GetType();
+            if (t == typeof(Directory_SymbolicLinks) || t == typeof(DirectoryInfo_SymbolicLinks))
+            {
+                File.CreateSymbolicLink(path, pathToTarget);
+            }
+            else
+            {
+                Directory.CreateSymbolicLink(path, pathToTarget);
+            }
+        }
+
+        /// <summary>Calls the actual public API for resolving the symbolic link target.</summary>
+        protected abstract FileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget = false);
+
+        /// <summary>
+        /// Asserts that the FullPath of the FileSystemInfo returned by ResolveLinkTarget() matches with the expected path of the file created.
+        /// Trims the Windows device prefix, in case there's any, before comparing.
+        /// </summary>
+        private static void AssertFullNameEquals(string expected, string actual)
+        {
+#if WINDOWS
```

Comment:
In the runtime repo we rather use `TARGET_WINDOWS` instead of `WINDOWS` (I could not find other usages of it in the repo)

```suggestion
#if TARGET_WINDOWS
```

but you might need to define it in both `System.IO.FileSystem.Tests.csproj` and `System.IO.FileSystem.Net5Compat.Tests.csproj`

Reply:
Here's one in src/**/tests/*.cs:
https://github.com/dotnet/runtime/blob/be80f675c7aa1fc895c053f80229d2591c69e5eb/src/installer/tests/HostActivation.Tests/PortableAppActivation.cs#L635-L636

Otoh, I couldn't find any usage of `TARGET_WINDOWS` in our tests.